### DISCUSSION TITLE
Email Address Variables

### DISCUSSION THREAD

**Discussion by ItsLeon15** [community/community#37408] [Copilot Conversations]
**Description**: Github Copilot is unable to complete a section of code if the code requires an email address to be entered.
**Steps to reproduce**: Create a basic function that requires an email variable as a string of which Github Copilot will be unable to complete it.
**Expected behavior**: Github Copilot should be able to recognize that an email address is required and should either enter the value from what it thinks is the best option or to leave it blank to be filled in by the user.

**Comment by DenverCoder1** [comment]
I think this is intentionally done in order to not accidentally leak random personal info into the project.

It behaves similarly around API tokens and most URLs.

**Reply by ItsLeon15** [reply]
That would make sense, but then couldn't the AI make the variable nothing instead of leaving it as `var email = ";`
`var email = " ";`

**Reply by DenverCoder1** [reply]
I agree, an empty spot would be fine. I think it would be nice to even leave a placeholder value eg. "example@example.com" in place of the email addresses so at least it can finish the rest of the completion.

I've found this issue to slow me down as well when using copilot.

**Reply by nalbion** [reply]
how do variables work? I've seen them mentioned in ` vscode.ChatAgentRequest` of the VS Code chatAgents API, but I haven't been able to figure out how to set a variable

**Comment by iamcjwilliams** [comment]
@ItsLeon15 @DenverCoder1 Thank you for your feedback, I will be sure our copilot team is notified

**Comment by omegasoft7** [comment]
@iamcjwilliams are there any updates regarding this issue?

**Comment by wolfgangmeyers** [comment]
It's incredibly annoying. It's not like I'm ever hard-coding email addresses - except in tests. Not having auto-complete in the tests ends up slowing things down with no added value.
